samples/excel/06-named-item/create-and-remove-named-item.yaml

```diff
@@ -0,0 +1,182 @@
+id: excel-06-named-item-create-and-remove-named-item-
+name: Create a named item
+description: Create a named item for a formula
+host: EXCEL
+api_set:
+    ExcelApi: 1.4
+script:
+    content: |
+        $("#add-name").click(addName);
+        $("#remove-name").click(removeName);
+        $("#setup").click(setup);
+
+        async function addName() {
+            try {
+                await Excel.run(async (context) => {
+
+                    const sheet = context.workbook.worksheets.getItem("Sample");
+
+                    sheet.names.add("TotalAmount", "=SUM(ExpensesTable[AMOUNT])");
+
+                    sheet.getRange("D11").values = [["=TotalAmount"]];
+
+                    sheet.getUsedRange().format.autofitColumns();
+                    sheet.getUsedRange().format.autofitRows();
+
+                    await context.sync();
+                });
+            }
+            catch (error) {
+                OfficeHelpers.UI.notify(error);
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        async function removeName() {
+            try {
+                await Excel.run(async (context) => {
+
+                    const sheet = context.workbook.worksheets.getItem("Sample");
+
+                    sheet.names.getItem("TotalAmount").delete();
+
+                    // Replace the named item (TotalAmount) with the actual formula for TotalAmount to avoid displaying #NAME in the cell.
+                    sheet.getRange("D11").values = [["=SUM(ExpensesTable[AMOUNT])"]];
+
+                });
+            }
+            catch (error) {
+                OfficeHelpers.UI.notify(error);
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        async function setup() {
+            try {
+                await Excel.run(async (context) => {
+
+                    await OfficeHelpers.ExcelUtilities.forceCreateSheet(context.workbook, "Sample");
+
+                    const sheet = context.workbook.worksheets.getItem("Sample");
+
+                    let expensesTable = sheet.tables.add("A1:D1", true);
+                    expensesTable.name = "ExpensesTable";
+
+                    expensesTable.getHeaderRowRange().values = [["Date", "Merchant", "Category", "Amount"]];
+
+                    let newData = Transactions.map(item =>
+                        [item.DATE, item.MERCHANT, item.CATEGORY, item.AMOUNT]);
+
+                    expensesTable.rows.add(null, newData);
+                    sheet.getUsedRange().format.autofitColumns();
+                    sheet.getUsedRange().format.autofitRows();
+                    sheet.activate();
+
+                    await context.sync();
+                });
+            }
+            catch (error) {
+                OfficeHelpers.UI.notify(error);
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        const Transactions = [
+            {
+                "DATE": "1/1/2017",
```

Comment:
I think you misunderstood my previous comment about naming the properties. I meant to say that the property names should be camel case instead of upper case: DATE -> date, MERCHANT -> merchant and so on.